kombu/pools.py

```diff
@@ -104,9 +105,18 @@ def register_group(group):
 class Connections(PoolGroup):
     """Collection of connection pools."""
 
-    def create(self, connection, limit):
+    def create(self, connection: Connection, limit: int | None):
         return connection.Pool(limit=limit)
 
+    def __getitem__(self, connection: Connection) -> ConnectionPool:
```

Comment:
I would like to know, what this dunder methods are going to achieve?

Reply:
I have made it very clear

Code editors such as vscode, pycharm, etc. cannot provide intelligent type inference for this connections[connection].acquire(block=True) code, so I would like to add typing hint to achieve this (intelligent type inference)

Reply:
When your mouse hovers over a conn, vscode will tell you that it is the type of this conn is Any, which is very stupid.

When hovering over a conn, I wish vscode would tell me what the explicit type is, instead of a stupid and damned Any

Reply:
And I added these typing hint which will solve this problem very well

Reply:
No type hint 👇
<img width="703" alt="图片" src="https://user-images.githubusercontent.com/38725104/196094045-2bd03fe3-5d76-4cb6-a778-08caedd82344.png">

With type annotations 👇
<img width="930" alt="图片" src="https://user-images.githubusercontent.com/38725104/196095775-3b4fca98-9045-4c8b-ad58-c3d217c17320.png">


Modern program development is inseparable from the intelligent hints provided by the IDE. With these type annotations, the IDE can provide us with type inference, allowing us to program more efficiently and enjoyably